plugins/kubernetes/app/decorators/build_decorator.rb

```diff
@@ -5,4 +5,31 @@
   def project_name
     @project_name ||= project.name.parameterize('-')
   end
+
+  def import_kubernetes_roles
+    # TODO : Build -> KubernetesRoles association
+    project.roles.destroy_all
+
+    config_files = project.repository.directory_contents(git_sha, path: 'kubernetes', ).select { |file|
+      file.ends_with?('.yml', '.yaml', '.json')
+    }
+
+    config_files.map do |file|
+      file_contents = project.repository.file_contents(git_sha, file)
+      config_file = Kubernetes::RoleConfigFile.new(file_contents, file)
+
+      # TODO : Build -> KubernetesRoles association
```

Comment:
Why do you need an association between the two?  They're connected via Build -> Release -> ReleaseDoc -> Role.